packages/compiler/core/checker.ts

```diff
@@ -516,7 +516,8 @@ export function createChecker(program: Program): Checker {
       | OperationStatementNode
       | UnionStatementNode
   ): number {
-    return node.symbol!.id!;
+    // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
+    return node.symbol?.id!;
```

Comment:
This is because node.symbol lies about being non-nullable? Do we not expect all of model, alias, interface, operation, union to have a node?

Reply:
the node types are correct, this method is what is cheating and why we had that crash before. Could try making it it valid(`return number | undefined`) but I think that affected a bit more code in the checker. Ultimately that would be better than cheating the type system and having those kind of issue keep coming back

Reply:
I guess nvm, the type is also wrong saying there there is always a symbol

Reply:
![image](https://user-images.githubusercontent.com/1031227/178338760-c737770a-ae2b-4b5e-91bc-f05a782f8ca5.png)

Reply:
We should maybe think about making this better in the typing, but that would be out of scope for this change, and likely very disruptive.